2019Labs/CustomSecurityContent/documentation/lab2_profiles.adoc

```diff
@@ -0,0 +1,403 @@
+
+Let’s imagine that one of the requirements set in your company policy is that the user sessions must timeout after 10 minutes of user’s inactivity.
+
+At this point, we can reveal to you that ComplianceAsCode already contains an implementation of this requirement in a form of a rule.  You now need to add this rule to your “travel” profile.
+
+However, the rule in ComplianceAsCode is generic, or in other words, customizable. It can check for arbitrary period of user’s inactivity. We need to set the specific value (10 minutes) in the profile.
+
+. Find the rule ID.
+** This is similar to the previous action.
++
+----
+git grep -i "Interactive Session Timeout" "*rule.yml"
+linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml:title: 'Set Interactive Session Timeout'
+----
++
+** As you already know from the first lab track, the rule is located in `linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml`.
+** It is easy to spot that the rule ID is `accounts_tmout` because the rule ID is the name of the directory where the rule is located.
+. Add the rule ID to selections list in your travel profile.
++
+----
+gedit rhel8/profiles/travel.profile
+----
++
+** Add `accounts_tmout` as a new item in selections list.
+** Make sure your indentation is consistent, use spaces. Make sure there is no trailing whitespace!
+. Check the rule contents to find out that there is a variable involved
++
+----
+gedit linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml
+----
++
+** From the rule contents you can clearly see that it is parametrized by the `variable var_accounts_tmout`.
+** Notice that the variable `var_accounts_tmout` is used in the description instead of exact value.
+** In the HTML guide, you will later see that `var_accounts_tmout` has been assigned a value.
+** The value is also automatically substituted into OVAL checks, Ansible Playbooks and the remediation scripts.
+. Check out the variable
++
+----
+find . -name var_accounts_tmout*
+gedit linux_os/guide/system/accounts/accounts-session/var_accounts_tmout.var
+----
++
+** The variable has multiple options, see the options list:
++
+----
+options:
+    30_min: 1800
+    10_min: 600
+    15_min: 900
+    5_min: 300
+    default: 600
+----
++
+** The `options` are defined as a YAML dictionary which maps keys to values. The YAML dictionary keys are used as selectors in ComplianceAsCode, the YAML dictionary values are concrete values that will be used in the checks. You use the selector to choose the value in the profile.
+** You can add a new key and value to the `options` dictionary if none of the values suits your needs.
+** We will add a new pair of variable name and selector into the profile. We will use the `10_min` selector to choose the 600 seconds.
+. Add the variable and the selector to the selections list in your `travel` profile
++
+----
+gedit rhel8/profiles/travel.profile
+----
```

Comment:
This is really difficult to read. Try to make it a text that can be followed, i.e. text containing proper sentences.